astropy/io/votable/tree.py

```diff
@@ -2828,7 +2828,7 @@ def _write_binary(self, mode, w, **kwargs):
                     for i, converter in fields_basic:
                         try:
                             chunk = converter(array_row[i], array_mask[i])
-                            assert type(chunk) == type(b'')
```

Comment:
I think it's better to use ``six.binary_type`` here rather than then original version. Anyway I moved it back from doing isinstance.